addons/sale/models/sale_order_line.py

```diff
@@ -310,7 +310,7 @@ def _compute_no_variant_attribute_values(self):
                 if ptav._origin not in valid_values:
                     line.product_no_variant_attribute_value_ids -= ptav
 
-    @api.depends('product_id')
+    @api.depends('product_id', 'invoice_lines.move_id.state')
```

Comment:
We don't want to recompute the SO line description when there are changes on the invoice.
We had an override of `action_post` to force the name recomputation, but it was removed in #150137.
Could you bring it back instead and add a test to make sure the name is recomputed when the invoice is posted ?

Reply:
Done!
Also Updated base to saas-16.4